<launch>
  <arg name="input_autoware_state" default="/autoware/state" />
  <arg name="input_driving_capability" default="/autoware/driving_capability" />
  <!-- To be replaced by ControlCommand -->
  <arg name="input_prev_control_command" default="/control/vehicle_cmd" />
  <arg name="input_current_gate_mode" default="/control/current_gate_mode" />
  <arg name="input_twist" default="/localization/twist" />

  <arg name="output_control_command" default="/system/emergency/control_cmd" />
  <arg name="output_shift" default="/system/emergency/shift_cmd" />
  <arg name="output_turn_signal" default="/system/emergency/turn_signal_cmd" />
  <arg name="output_is_emergency" default="/system/emergency/is_emergency" />

  <arg name="service_clear_emergency" default="/system/clear_emergency" />

  <arg name="config_file" default="$(find-pkg-share emergency_handler)/config/emergency_handler.param.yaml" />

  <node pkg="emergency_handler" exec="emergency_handler" name="emergency_handler" output="screen">
    <remap from="input/autoware_state" to="$(var input_autoware_state)"/>
    <remap from="input/driving_capability" to="$(var input_driving_capability)"/>
    <remap from="input/prev_control_command" to="$(var input_prev_control_command)"/>
    <remap from="input/current_gate_mode" to="$(var input_current_gate_mode)"/>
    <remap from="input/twist" to="$(var input_twist)"/>

    <remap from="output/control_command" to="$(var output_control_command)"/>
    <remap from="output/shift" to="$(var output_shift)"/>
    <remap from="output/turn_signal" to="$(var output_turn_signal)"/>
    <remap from="output/is_emergency" to="$(var output_is_emergency)"/>

    <remap from="service/clear_emergency" to="$(var service_clear_emergency)"/>

    <param from="$(var config_file)" />
  </node>
</launch>
